spec/factories/projects.rb

```diff
@@ -0,0 +1,36 @@
+FactoryBot.define do
+  factory :project do
+    transient do
+      # The name of the host genome to create automatically for each sample
+      host_genome_name { nil }
+      # Array of samples entries to create.
+      # The hash elements will be passed on to sample factory as keyword arguments.
+      samples_data { [] }
+    end
+
+    sequence(:name) { |n| "Project #{n}" }
+    public_access { 0 }
+
+    trait :with_samples do
```

Comment:
I have a suggestion here: shouldn't we call this traits `with_sample` instead of `with_samples`, since it returns only one sample? We can apply same logic to `with_public_samples`